syntax = "proto2";

package org_tensorflow_demo;

// Prior for a single feature (like minimum x coordinate, width, area, etc.)
message BoxCoderPrior {
  optional float mean = 1 [default = 0.0];
  optional float stddev = 2 [default = 1.0];
};

// Box encoding/decoding configuration for a single box.
message BoxCoderOptions {
  // Number of priors must match the number of values used to encoded
  // values which is derived from the use_... flags below.
  repeated BoxCoderPrior priors = 1;

  // Minimum/maximum X/Y of the four corners are used as features.
  // Order: MinX, MinY, MaxX, MaxY.
  // Number of values: 4.
  optional bool use_corners = 2 [default = true];

  // Width and height of the box in this order.
  // Number of values: 2.
  optional bool use_width_height = 3 [default = false];

  // Coordinates of the center of the box.
  // Order: X, Y.
  // Number of values: 2.
  optional bool use_center = 4 [default = false];

  // Area of the box.
  // Number of values: 1.
  optional bool use_area = 5 [default = false];
};

// Options for MultiBoxCoder which is a encoder/decoder for a fixed number of
// boxes.
// A list of BoxCoderOptions that allows for storing multiple box coder options
// in a single file.
message MultiBoxCoderOptions {
  repeated BoxCoderOptions box_coder = 1;
};
